src/main/java/com/marklogic/contentpump/RDFReader.java

```diff
@@ -1075,29 +1046,25 @@ public boolean failed() {
             return failed;
         }
 
-        @Override
-        public void run() {
-            LangRIOT parser;
-            try {
-                parser = RiotReader.createParser(in, lang, fsname, 
-                        rdfInputStream);
-            } catch (Exception ex) {
-                // Yikes something went horribly wrong, bad encoding maybe?
-                LOG.error("Failed to parse(please check intactness and encoding): " + origFn, ex);
 
-                byte[] b = new byte[0] ;
-                InputStream emptyBAIS = new ByteArrayInputStream(b) ;
-                parser = RiotReader.createParser(emptyBAIS, lang, fsname, 
-                        rdfInputStream);
-            }
+        public void run() {
             try {
-                ErrorHandler handler = new ParserErrorHandler(fsname);
+                ErrorHandler handler =  ErrorHandlerFactory.errorHandlerStd(ErrorHandlerFactory.stdLogger);
                 ParserProfile prof = RiotLib.profile(lang, fsname, handler);
-                parser.setProfile(prof);
-                parser.parse();
+                if (lang == Lang.TRIG) {
+                    rdfInputStream = AsyncParser.of(in, lang, fsname).streamQuads();
```

Comment:
Could you please explain why we are using different apis for TRIG and NQUADs?

Reply:
Just the AsyncParser.asyncParse(in,lang,fsname,dest) should be able to handle all the types of RDFs. And this API call is currently being used for our in-memory RDF processing. The basic call asyncparse writes to streamRDF "dest" which is an event-based output. 

But our streamed RDF code flow is based on the parser output written and iterated via a stream. So to make minimal changes in the code flow I chose to use the stream-based API calls. 

We use 4 different API calls for the streaming RDF processing. For Ntriples and Nquads, the different API calls were suggested by the [Jena documentation](https://jena.apache.org/documentation/io/rdf-input.html#iterating-over-parser-output). For the Trig and else part, they are the same API calls but the difference is what they emit after parsing. Trig file outputs are emitted as quads and everything else like ttl, json, RDFXML file outputs are emitted as triples. Hence same API calls with different stream calls ( streamQuads()/streamTriples() )